File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

int solution(vector<int> elements) {
    set<int> sums;
    int size = elements.size();

    for (int i = 0; i < size; i++) {
        int sum = 0;
        for (int j = i; j < i + size; j++) {
            sum += elements[j % size];
            sums.emplace(sum);
        }
    }

    return sums.size();
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.swift
import Foundation

func solution(_ elements:[Int]) -> Int {
    var answer: Set<Int> = []
    
    for i in 0..<elements.count {
        var sum: Int = 0
        for j in i..<(i+elements.count) {
            let index = j % elements.count
            sum += elements[index]
            answer.insert(sum)
        }
    }
    return answer.count
}